crates/vault-registry/src/lib.rs

```diff
@@ -86,8 +85,12 @@ pub mod pallet {
     use frame_support::pallet_prelude::*;
     use frame_system::pallet_prelude::*;
 
+    /// The current storage version.
+    const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
     #[pallet::pallet]
     #[pallet::generate_store(trait Store)]
+    #[pallet::storage_version(STORAGE_VERSION)]
```

Comment:
What does this do, exactly? Does it set it to 0 when building genesis, or does it set the default to 1? What I'm saying is, kintsugi and interlay might actually get set to 0..

 Also, the whole situation with `palletVersion`/`storageVersion` is a bit of a mess. I thought we'd use `palletVersion`, since that storage item already exists on polkadotjs. But then I found [this pr](https://github.com/paritytech/substrate/pull/9165), which changed the old `palletVersion` to `storageVersion`. It turns out that the palletVersion is no longer used and that the item shown by polkadotjs is actualy the storageVersion: https://github.com/polkadot-js/api/issues/546.

Reply:
@sander2 & @gregdhill 

---

When performing an upgrade, we need to obtain the values for `current_storage_version` and `on_chain_storage_version`.

During the genesis build, we set the value for `on_chain_storage_version`, and in the macro, we set the value for `current_storage_version`.

For the runtime upgrade:
- Retrieve both values and compare them.
- In the code before the runtime upgrade Increment `current_storage_version` by one ie is the `STORAGE_VERSION` constant value.
- If the migration is successful, increment `on_chain_storage_version` by one as well.
- Both the `on_chain_storage_version` and `current_storage_version` should be the same if the runtime upgrade is successful. 
- But since we already have performed multiple runtime upgrades we need to make sure that `on_chain_storage_version` matches the `current_storage_version`. [PR 1106](https://github.com/interlay/interbtc/pull/1106) fixes it. 

References:
- Stack Exchange: [On-chain storage version on new chain seems to be incorrect](https://substrate.stackexchange.com/questions/3146/on-chain-storage-version-on-new-chain-seems-to-be-incorrect)
- Parity Documentation: [GetStorageVersion trait - current_storage_version](https://paritytech.github.io/substrate/master/frame_support/traits/trait.GetStorageVersion.html#tymethod.current_storage_version)

---